### ISSUE TITLE
java.lang.UnsatisfiedLinkError: Error looking up function 'gettid': java: undefined symbol: gettid

### ISSUE THREAD

**Issue by silencewood** [oshi/oshi#2389]
oshi version is 6.4.2
jna version is 5.13.0
cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 6.5 (Santiago)

Exception in thread "main" java.lang.UnsatisfiedLinkError: Error looking up function 'gettid': java: undefined symbol: gettid
        at com.sun.jna.Function.<init>(Function.java:252)
        at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:620)
        at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:596)
        at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:582)
        at com.sun.jna.Library$Handler.invoke(Library.java:248)
        at com.sun.proxy.$Proxy1.gettid(Unknown Source)
        at oshi.software.os.linux.LinuxOperatingSystem.getThreadId(LinuxOperatingSystem.java:265)

**Comment by dbwiddis** [comment]
OSHI is looking for the [`gettid()`](https://man7.org/linux/man-pages/man2/gettid.2.html) function in your C library and not finding it.

This function is in glibc 2.3 and later.  It was added to musl libc in 1.2.2.

What is your libc version?

**Comment by dbwiddis** [comment]
> Red Hat Enterprise Linux Server release 6.5 (Santiago)

RHEL 6.5 was released in 2013; the latest 6.x is 6.10.  Looks like it's in "extended support" phase with EOL in 2024.

RHEL 6 uses an older version of glibc that doesn't have `gettid()`.

TLDR: OSHI's thread code above is not compatible with very old versions of RHEL/glibc.  

If you'd like to contribute a PR to fetch the information another way to support your old OS, it'd be welcome.

**Comment by silencewood** [comment]
Yes,is libc version problem

[root@test1 bin]# cd /bin
[root@test1 bin]# ldd ls
        linux-vdso.so.1 =>  (0x00007fffb0dff000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x0000003536c00000)
        librt.so.1 => /lib64/librt.so.1 (0x0000003535c00000)
        libcap.so.2 => /lib64/libcap.so.2 (0x0000003539000000)
        libacl.so.1 => /lib64/libacl.so.1 (0x0000003540000000)
        libc.so.6 => /lib64/libc.so.6 (0x0000003535400000)
        libdl.so.2 => /lib64/libdl.so.2 (0x0000003535000000)
        /lib64/ld-linux-x86-64.so.2 (0x0000003534c00000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x0000003535800000)
        libattr.so.1 => /lib64/libattr.so.1 (0x0000003545c00000)
[root@test1 bin]# ls -l /lib64/libc.so.6
lrwxrwxrwx. 1 root root 12 Dec  4 17:09 /lib64/libc.so.6 -> libc-2.12.so
[root@test1 bin]# rpm -qa|grep glibc
glibc-devel-2.12-1.132.el6.x86_64
glibc-2.12-1.132.el6.x86_64
glibc-common-2.12-1.132.el6.x86_64
glibc-headers-2.12-1.132.el6.x86_64

**Comment by dbwiddis** [comment]
It's apparently not a good idea to change glibc versions in your RHEL install, so your choices are:
 - upgrade to a more recent (or less recent!) version of RHEL that has glibc 2.3 support
 - don't use the thread functions in OSHI
 - find another way to get the thread info and submit a PR for OSHI to be compatible with this OS
